test/addons-napi/test_array/test.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+const common = require('../../common');
+const assert = require('assert');
+
+// Testing api calls for arrays
+const test_array = require(`./build/${common.buildType}/test_array`);
+
+const array = [
+  1,
+  9,
+  48,
+  13493,
+  9459324,
+  { name: 'hello' },
+  [
+    'world',
+    'node',
+    'abi'
+  ]
+];
+
+assert.strictEqual(test_array.Test(array, array.length + 1),
+                   'Index out of bound!');
+
+try {
+  test_array.Test(array, -2);
+} catch (err) {
+  assert.strictEqual(err.message, 'Invalid index. Expects a positive integer.');
+}
```

Comment:
Fixed in https://github.com/nodejs/node/pull/11975/commits/9c0b1517b894d0a655377d23f9e4b9943d93aa38